Water Rendering

VectorIV opened the pull request:
Added simple water rendering overview page including figure images.

feliwir committed:
Prepare water rendering post

Commit:
Added simple water rendering overview page including figure images.

feliwir commented:
@VectorIV could you export the Photoshop files as PNG and include them, since JPG tends to be a bit blurry for lineart. Also please remove the original PSD files if possible. Apart from that it looks great :+1:

feliwir commented on .vs/VSWorkspaceState.json:
```diff
@@ -0,0 +1,10 @@
+{
```
Please don't commit your ".vs" folder

Commit:
-Remove .psd files
-Convert viable files to .png
-Added conclusive result paragraph and final image

Commit:
-Removed  .vs files

Skleni commented on posts/water_rendering/water_rendering.md:
```diff
@@ -0,0 +1,71 @@
+*<center>Figure 2: Camera For Refraction Rendering</center>*
+
+## Waves
+This is done at shader level either via vertex or fragment. Many techniques could be used here including DuDv mapping, bump mapping, nomral mapping, etc.
```
Typo: nomral

Skleni commented on posts/water_rendering/water_rendering.md:
```diff
@@ -0,0 +1,71 @@
+This is done at shader level either via vertex or fragment. Many techniques could be used here including DuDv mapping, bump mapping, nomral mapping, etc.
+We'll using only fragment for this implementation.
+
+**DuDv Mapping** is simply a directoinal distortion map usually derived from a normal map. It only has R and G channels for offsetting U and V vaules of the texture. This creates ripple effect on the surface.
```
Typos: directoinal, vaules, effect(s?)

Skleni commented on posts/water_rendering/water_rendering.md:
```diff
@@ -0,0 +1,71 @@
+![DuDv Map](./wdudv.jpg)
+*<center>Figure 3: Sample DvDu Map</center>*
+
+**Normal Mapping** is an extensive bump map uses for lighting calculation. Having all RGB channels, they are represented as XYZ directions. This creates the illusion of detailed surface without having a lot of polygons.
```
used insted of uses?
...of _a_ detailed surface?

Skleni commented on posts/water_rendering/water_rendering.md:
```diff
@@ -0,0 +1,71 @@
+
+## Caustics
+This is also done at shader level as well using fragment. The simplest and the only way I know how to do is using animated textures. By having a set of pre-baked textures, one can animate it to simulate the effects of caustics.
+At higher level of shadering, I assume one may simulate casutics from the movement of the waves and direction of the lightsource.
```
Typo: casutics

Skleni commented on posts/water_rendering/water_rendering.md:
```diff
@@ -0,0 +1,71 @@
+*<center>Figure 6: Caustics Rendering</center>*
+
+## Miscellaneous
+**Fresnel Effect** A principle of reflectivity. Shallow angle = strong reflectivity, steep nagle weak reflectivity. By applying this principle to reflection and refraction rendering, one can improve the realism of water rendering.
```
Typo: nagle
missing = before weak?

Skleni commented on posts/water_rendering/water_rendering.md:
```diff
@@ -0,0 +1,71 @@
+## Result
+Combining those components together, you get the result down below: an improved version of vanilla water from early reveal screenshots. While not the fanciest water you might have ever seen but still, it looks more watery than AoE4's attempt.
+![Result](./result.png)
+With time and afford, you can make it even more convincing but that is a discussion for another time.
```
effort?

Skleni reviewed (changes requested):
Found some typos...otherwise, great work! :)

Skleni commented:
Why are the roads so broken in the result image? Is this based on the current codebase?

Commit:
-Fixed typos

Skleni reviewed (approved):
Thanks!

feliwir committed:
Update to latest packages
